tests/ssl-opt.sh

```diff
@@ -656,14 +656,32 @@ fi
 # used by watchdog
 MAIN_PID="$$"
 
-# be more patient with valgrind
+# We use somewhat arbitrary delays for tests:
+# - how long do we wait for the server to start (when lsof not available)?
+# - how long do we wait for the client to finish?
+#   (not to check performance, just to avoid waiting indefinitely)
+# - how long do we wait after client exit for completion of server tasks?
+#   (in particular, for it to finish writing & flushing its log file)
+# Things are slower with valgrind, so give extra time here.
+#
+# Note: there is a trade-off between the running time of this script and the
+# risk of spurious errors because we didn't wait long enough. In a better
+# world, we would be able to do without those delays (except the watchdog).
 if [ "$MEMCHECK" -gt 0 ]; then
-    START_DELAY=3
-    DOG_DELAY=30
+    START_DELAY=6
+    DOG_DELAY=60
+    SRV_DELAY_BASE=2
```

Comment:
Should impatient devs be able to override the SRV_DELAY_BASE when valgrind is being used?

Reply:
Impatient devs on Linux are likely to prefer ASan plus MemSan to valgrind, because even with the need to do two builds, it ends up being faster (and last time I checked, checking more) that valgrind's memcheck.

Impatient devs on OS X are out of luck because MemSan is still not available on non-Linux, and last time I tried (but this was two years ago), `ssl-opt.sh --memcheck` didn't work on OS X because apparently processes running under valgrind where unable to receive/catch signals, and we rely on the server cleaning up after receiving SIGTERM for our memory tests in `ssl-opt.sh` and `compat.sh`.

Now, of course it wouldn't hurt to make that overrideable too, if you think it can be useful (or better for consistency).

Reply:
Impatient devs have been known to edit those timeouts down while they were developing. Ideally the scripts would be smarter and timeouts would only be needed in case a bug leads to an infinite loop. But I think this ideal world will wait for the day we rewrite those scripts in Python.